scripts/test.sh

```diff
@@ -81,26 +102,118 @@ start_packager() {
+    export IOS_SIM_DEVICE
+  done
+  if [ -z "$IOS_SIM_DEVICE" ]; then
+    echo "*** Failed to determine the iOS Simulator device to use ***"
+    exit 1
+  fi
+  
+  # - Wait until the simulator is fully booted by waiting for it to launch SpringBoard
+  printf "Waiting for springboard to ensure device is ready..."
+  xcrun simctl launch "$IOS_SIM_DEVICE" com.apple.springboard >/dev/null 2>&1 || true
+  echo "  done"
+  
+  # - Run the build and test
   if [ -n "$XCPRETTY" ]; then
-    LOGTEMP=`mktemp build.log.XXXXXX`
-    trap "rm $LOGTEMP" EXIT
-    xcodebuild -scheme "$1" -configuration "$CONFIGURATION" -sdk iphonesimulator -destination id="$dest" build test 2>&1 | tee "$LOGTEMP" | "$XCPRETTY" -c --no-utf --report junit --output build/reports/junit.xml || {
+    log_temp=`mktemp build.log.XXXXXX`
+    if [ -e "$log_temp" ]; then
+      rm "$log_temp"
+    fi
+    xcrun xcodebuild -scheme "$1" -configuration "$CONFIGURATION" -sdk iphonesimulator -destination name="iPhone 5s" build test 2>&1 | tee "$log_temp" | "$XCPRETTY" -c --no-utf --report junit --output build/reports/junit.xml || {
 	    EXITCODE=$?
         printf "*** Xcode Failure (exit code $EXITCODE). The full xcode log follows: ***\n\n"
-        cat "$LOGTEMP"
+        cat "$log_temp"
         printf "\n\n*** End Xcode Failure ***\n"
         exit $EXITCODE
     }
+    rm "$log_temp"
   else
-    xcodebuild -scheme "$1" -configuration "$CONFIGURATION" -sdk iphonesimulator -destination id="$dest" build test || {
+    xcrun xcodebuild -scheme "$1" -configuration "$CONFIGURATION" -sdk iphonesimulator -destination id="$IOS_SIM_DEVICE" build test || {
 	    EXITCODE=$?
         echo "*** Failure (exit code $EXITCODE). ***"
         exit $EXITCODE
-	}
+	  }
   fi
 }
 
+setup_ios_simulator() {
+  # - Ensure one version of xcode is chosen by all tools
+  
+  if [[ -z "$DEVELOPER_DIR" ]]; then
+    export DEVELOPER_DIR="$(xcode-select -p)"
+  fi
+  
+  # - Kill the Simulator to ensure we are running the correct one
+  
+  echo "Resetting simulator using toolchain from: $DEVELOPER_DIR"
+  
+  # quit Simulator.app
+  local deadline=$((SECONDS+5))
+  while pgrep -qx Simulator && [ $SECONDS -lt $deadline ]; do
+    osascript -e 'tell app "Simulator" to quit without saving' || true
+    sleep 0.25 # the 
+  done
+  
+  # stop CoreSimulatorService
+  launchctl remove com.apple.CoreSimulator.CoreSimulatorService 2>/dev/null || true
+  sleep 0.25 # launchtl can take a small moment to kill services
+  
+  # kill them with fire
+  while pgrep -qx Simulator com.apple.CoreSimulator.CoreSimulatorService; do
+    pkill -9 -x Simulator com.apple.CoreSimulator.CoreSimulatorService || true
+    sleep 0.05
+  done
+  
+  # - Prod `simctl` a few times to make sure it is ready to go
```

Comment:
why? what happens when it's not ready to go? is there a better metric than running this continuously for 5 seconds to indicate when it's "ready"?